ykcompile/src/lib.rs

```diff
@@ -251,11 +257,77 @@ impl<TT> TraceCompiler<TT> {
                     ; mov [rdi + aoff], rax
                 );
             }
+            (Location::Register(reg), Location::Deref(boxed)) => match *boxed {
+                Location::Stack(off) => {
+                    dynasm!(self.asm
+                        ; mov Rq(reg), [rbp - off]
+                        ; mov Rq(reg), [Rq(reg)]
+                    );
+                }
+                Location::Register(reg2) => {
+                    dynasm!(self.asm
+                        ; mov Rq(reg), [Rq(reg2)]
+                    );
+                }
+                _ => todo!(),
+            },
+            (Location::Stack(off1), Location::Deref(boxed)) => match *boxed {
+                Location::Stack(off2) => {
+                    dynasm!(self.asm
+                        ; mov rax, [rbp - off2]
+                        ; mov rax, [rax]
+                        ; mov [rbp - off1], rax
+                    );
+                }
+                Location::Register(reg) => {
+                    dynasm!(self.asm
+                        ; mov rax, [Rq(reg)]
+                        ; mov [rbp - off1], rax
+                    );
+                }
+                _ => todo!(),
+            },
             _ => unreachable!(),
         }
         Ok(())
     }
 
+    fn mov_place_ref(&mut self, p1: &Place, p2: &Place) -> Result<(), CompileError> {
+        let lloc = self.place_to_location(p1)?;
+        // We can only reference Locals living on the stack. So move it there if it doesn't.
+        let rloc = match self.place_to_location(p2)? {
+            Location::Register(reg) => {
+                let offset = self.stack_builder.alloc(8, 8) as i32;
```

Comment:
(8, 8) here is the size and align of a pointer? I wonder if it should be a constant?

Reply:
Dunno, haven't really been thinking about the type stuff yet, so just copy-pasted from above. We can fix that when we do the type changes?

Reply:
ok